app/services/transaction_service/gateway/paypal_adapter.rb

```diff
@@ -65,7 +65,7 @@ def get_payment_details(tx:)
 
       payment_total = payment[:payment_total].or_else(nil)
       total_price = Maybe(payment[:payment_total].or_else(payment[:authorization_total].or_else(nil)))
-                    .or_else(tx[:unit_price])
+                    .or_else(tx[:unit_price] * tx[:listing_quantity])
```

Comment:
I have a feeling this is the wrong place to default to tx based calculation. We are in payment service so the data returned is intentionally purely payment based. On top of this there's transaction_service/transaction.rb that return the final tx response. On line 183 there method to_tx_response already adds item_total into the response that is calculated using this exact logic. So ultimately defaulting to showing the item_total in place of actual payment total is the responsibility of the view (where it's possible also to tell the user that this is not the final payment price and only an estimate because we don't know all the details before an actual payment is made).